lib/product_list/cubits/product_search_cubit.dart

```diff
@@ -54,4 +59,38 @@ class ProductSearchCubit extends Cubit<ProductSearchState> {
       ));
     }
   }
+
+  Future<void> nextPage() async {
+    if (state.pagination.currentPage + 1 >= state.pagination.totalPages ||
+        state.searchResults == null ||
+        state.status == Status.pending) {
+      return;
+    }
+
+    try {
+      // todo: discuss how to handle item pending state
+      // emit(state.copyWith(status: Status.pending));
```

Comment:
Either we consider:
- Pending status and no data == initial loading
- Pending status and some data already there == page loading

Or we can have a `pagePending` status for a more descriptive state (I like this solution better)